From 1899cbe609a3bedc8cb1da229c69cce9a09da2b5 Mon Sep 17 00:00:00 2001
From: Mike Dunn <mikedunn@newsguy.com>
Date: Sun, 11 Mar 2012 14:21:10 -0700
Subject: [PATCH] mtd: add ecc_strength fields to mtd structs

This adds 'ecc_strength' to struct mtd_info.  This stores the maximum number of
bit errors that can be corrected in one writesize region.

For consistency with the nand code, 'strength' is similiarly added to struct
nand_ecc_ctrl.  This stores the maximum number of bit errors that can be
corrected in one ecc step.

Signed-off-by: Mike Dunn <mikedunn@newsguy.com>
Signed-off-by: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
(cherry picked from commit 1d0b95b0834087ba3653f69c24483d63a26d51a7)
---
 include/linux/mtd/mtd.h  | 3 +++
 include/linux/mtd/nand.h | 2 ++
 2 files changed, 5 insertions(+)

--- a/include/linux/mtd/mtd.h
+++ b/include/linux/mtd/mtd.h
@@ -140,6 +140,9 @@ struct mtd_info {
 	/* ecc layout structure pointer - read only ! */
 	struct nand_ecclayout *ecclayout;
 
+	/* max number of correctible bit errors per writesize */
+	unsigned int ecc_strength;
+
 	/* Data for variable erase regions. If numeraseregions is zero,
 	 * it means that the whole device has erasesize as given above.
 	 */
--- a/include/linux/mtd/nand.h
+++ b/include/linux/mtd/nand.h
@@ -331,6 +331,7 @@ struct nand_hw_control {
  * @steps:	number of ecc steps per page
  * @size:	data bytes per ecc step
  * @bytes:	ecc bytes per step
+ * @strength:	max number of correctible bits per ECC step
  * @total:	total number of ecc bytes per page
  * @prepad:	padding information for syndrome based ecc generators
  * @postpad:	padding information for syndrome based ecc generators
@@ -356,6 +357,7 @@ struct nand_ecc_ctrl {
 	int size;
 	int bytes;
 	int total;
+	int strength;
 	int prepad;
 	int postpad;
 	struct nand_ecclayout	*layout;
